File: spring-security/src/main/java/com/example/spring_security/dto/PropertyListingDTO.java
package com.example.spring_security.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PropertyListingDTO {
    private Long id;
    private String title;
    private String description;
    private BigDecimal price;
    private String location;
    private Double area;
    private String propertyType;
    private String contactInfo;
    private String city;
    private String district;
    private String fullAddress;
    private Double latitude;
    private Double longitude;
    private String formattedAddress;
    private Integer rooms;
    private Integer bedrooms;
    private Integer bathrooms;
    private List<String> imageUrls;
    private String sourceUrl;
    private String sourceWebsite;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private LocalDateTime listingDate;
    private boolean active;
    
    // Owner-specific fields
    private BigDecimal securityDeposit;
    private LocalDate availableTo;
    private String paymentFrequency;
    private Integer minimumStayMonths;
    private Boolean hasBalcony;
    private Integer floor;
    private Set<String> amenities;
    
    // Source type and owner information
    private String sourceType; // "OWNER" or "SCRAPED"
    private Long ownerId; // ID of the owner if sourceType is "OWNER"
} 

File: spring-security/frontend/ui/src/app/models/property-listing.dto.ts
export interface PropertyListingDTO {
  id: number;
  title: string;
  description: string;
  price: number;
  location: string;
  area: number;
  propertyType: string;
  contactInfo: string;
  city: string;
  district: string;
  fullAddress: string;
  latitude: number;
  longitude: number;
  formattedAddress: string;
  rooms: number;
  bedrooms: number;
  bathrooms: number;
  imageUrls: string[];
  sourceUrl: string;
  sourceWebsite: string;
  createdAt: string;
  updatedAt: string;
  listingDate: string;
  active: boolean;
  mainImageUrl?: string;
  
  // Owner-specific fields
  securityDeposit: number;
  availableTo: string;
  paymentFrequency: string;
  minimumStayMonths: number;
  hasBalcony: boolean;
  floor: number;
  amenities: string[];
  
  // Source type and owner information
  sourceType: string; // "OWNER" or "SCRAPED"
  ownerId: number;
} 